plugins/microservices/msi_get_open_data_obj_l1desc_index/libmsi_get_open_data_obj_l1desc_index.cpp

```diff
@@ -0,0 +1,83 @@
+/// \file
+
+#include "irods_ms_plugin.hpp"
+#include "irods_re_structs.hpp"
+#include "msParam.h"
+#include "rodsErrorTable.h"
+#include "irods_error.hpp"
+#include "irods_log.hpp"
+#include "objDesc.hpp"
+#include "fileOpr.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+// persistent L1 object descriptor table
+extern l1desc_t L1desc[NUM_L1_DESC];
+
+namespace
+{
+    int msi_impl(msParam_t* _in, msParam_t* _out, ruleExecInfo_t* _rei)
+    {
+        // check sanity
+        if (!_in || !_out) {
+            return SYS_INTERNAL_NULL_INPUT_ERR;
+        }
+
+        // input param needs to be a string
+        if (strcmp(_in->type, STR_MS_T)) {
+            return SYS_INVALID_INPUT_PARAM;
+        }
+
+        const char *objPath = (const char*)_in->inOutStruct;
+        int l1descInx = -1;
+
+        for (const l1desc_t &l1 : L1desc)
+        {
+            // for a valid descriptor, if the path matches...
+            if (l1.inuseFlag == FD_INUSE && !strcmp(l1.dataObjInp->objPath, objPath)) {
+                l1descInx = &l1 - L1desc;
+            }
+        }
```

Comment:
should a `break` statement be added to the body of the if-statement?

Reply:
Fixed. This was probably an error in my thought process. Just thought in case the iRODS API allows for the same client / agent to open up many L1 descriptors to the same object, Like in POSIX I believe a process can have many `fd`'s pointing to the same file with different current offsets and flags. So for some reason I thought that it would be "best" to return the last L1desc index found, not the first. Getting the first is certainly faster. 

Seems like the same logic was followed here https://github.com/jasoncoposky/irods_policy_composition_framework/blob/master/policy_composition_framework_utilities.cpp#L213 to which I was pointed to as an example.